src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,523 @@
+			return false;
+		}
+
+		// If the plugin is not activated, activate it.
+		if ( ! $dependency_is_active ) {
+			$this->add_notice_activate( get_plugin_data( WP_PLUGIN_DIR . '/' . $plugin ), $dependency );
+			return false;
+		}
+
+		// Add item to the $dependencies_parents array.
+		if ( empty( $this->dependencies_parents[ $dependency->file ] ) ) {
+			$this->dependencies_parents[ $dependency->file ] = array();
+		}
+		$this->dependencies_parents[ $dependency->file ][] = $plugin;
+
+		return true;
+	}
+
+	/**
+	 * Add notices.
+	 *
+	 * @since 5.9.0
+	 * @access public
+	 *
+	 * @return void
+	 */
+	public function admin_notices() {
+		// Early return if there are no notices to display.
+		if ( empty( $this->notices ) ) {
+			return;
+		}
+
+		foreach ( $this->notices as $notice ) {
+			echo '<div class="notice notice-warning plugin-dependencies"><p>' . wp_kses_post( $notice['content'] ) . '</p></div>';
+		}
+	}
+
+	/**
+	 * Cancel plugin's activation request.
+	 *
+	 * @since 5.9.0
+	 * @access public
+	 *
+	 * @return void
+	 */
+	public function cancel_activation_request() {
+		if ( empty( $_GET['action'] ) || 'cancel-activate' !== $_GET['action'] || empty( $_GET['plugin'] ) ) {
+			return;
+		}
+		$file = sanitize_text_field( wp_unslash( $_GET['plugin'] ) );
+		check_admin_referer( 'cancel-activate-plugin_' . $file );
+
+		$this->remove_plugin_from_queue( $file );
+	}
+
+	/**
+	 * Filters the action links displayed for each plugin in the Plugins list table.
+	 *
+	 * @since 5.9.0
+	 * @access public
+	 *
+	 * @param string[] $actions     An array of plugin action links.
+	 * @param string   $plugin_file Path to the plugin file relative to the plugins directory.
+	 * @param array    $plugin_data An array of plugin data. See `get_plugin_data()`.
+	 *
+	 * @return string[]
+	 */
+	public function plugin_action_links( $actions, $plugin_file, $plugin_data ) {
+
+		// Remove deactivation link from dependencies.
+		if ( ! empty( $this->dependencies_parents[ $plugin_file ] ) ) {
+			unset( $actions['deactivate'] );
+		}
+
+		// On plugins with unmet dependencies that the user has already requested for the plugin's activation,
+		// removes the activation link from its actions and add a "Cancel pending activation" link in its place.
+		if ( in_array( $plugin_file, $this->get_plugins_to_activate(), true ) && ! empty( $this->get_plugin_dependencies( $plugin_file ) ) ) {
+			unset( $actions['activate'] );
+			if ( current_user_can( 'activate_plugin', $plugin_file ) ) {
+				$cancel_activation = sprintf(
+					'<a href="%s" class="cancel-activate unmet-dependencies" aria-label="%s">%s</a>',
+					wp_nonce_url( 'plugins.php?action=cancel-activate&amp;plugin=' . rawurlencode( $plugin_file ), 'cancel-activate-plugin_' . $plugin_file ),
+					/* translators: %s: Plugin name. */
+					esc_attr( sprintf( _x( 'Cancel activation of %s', 'plugin' ), get_plugin_data( WP_PLUGIN_DIR . '/' . $plugin_file )['Name'] ) ),
```

Comment:
I suppose this could be simplified if the plugin name was always visible on the link text, then the aria-label would not be needed.

Reply:
This link is an action right below the plugin name. The plugin name is right above it, and so it follows the same pattern as all the other plugin-action links. We have "Activate" with an aria-label, deactivate & delete too. This just follows the same pattern as all other core links in that area.